server/pages/index.js

```diff
@@ -374,22 +377,57 @@ module.exports = function() {
 	} );
 
 	if ( config.isEnabled( 'manage/themes/details' ) ) {
+		function updateRenderCache( themeSlug ) {
+			let themeData = themeDetails.get( themeSlug );
+			wpcom.undocumented().themeDetails( themeSlug, ( error, data ) => {
+				if ( error ) {
+					console.log( `Error fetching theme ${ themeSlug } details: `, error.message || error );
+				} else if ( ! themeDetails.has( themeSlug ) || ( Date( data.date_updated ) > Date( themeData.date_updated ) ) ) {
+					console.log( 'caching', themeSlug );
+					themeDetails.set( themeSlug, data );
+					renderThemeSheet( data );
```

Comment:
Yep. It's kind of confusing, hence my comment about using a more explicit cache.